application/artifacts/artifacts_view.py

```diff
@@ -179,16 +180,53 @@ def patch(self, **params):
     def _upload_file(self, params):
         file_saver = FileSaver(params['file'])
         file_saver.save()
+
+        image_resizer = ImageResizer(file_saver.file_path)
+        image_resizer.save_sizes()
+
         return file_saver.get_metadata()
 
+class ImageResizer: #pylint:disable=too-few-public-methods
+    """Saves resized versions of an image to disk"""
+
+    widths = [320, 480, 640, 750, 1080]
+
+    def __init__(self, file_path):
+        self.image = Image.open(file_path)
+        self.original_file_name = file_path.split("/")[-1]
+
+    def save_sizes(self):
+        """Saves resized image for all defined widths"""
+        for width in self.widths:
+            height = self._calculate_height(width)
+            resized_image = self.image.resize((width, height), Image.ANTIALIAS)
+            resized_image.save(self._generate_file_path(width))
+            resized_image.close()
+
+    def _generate_file_path(self, width):
+        file_name = self._generate_file_name(width)
+        return os.path.join(current_app.config["UPLOAD_FOLDER"], file_name)
+
+    def _generate_file_name(self, width):
+        """Generate filename for a resized image of given width"""
+        [prefix, suffix] = self.original_file_name.rsplit('.', 1)
+        filename = f"{prefix}_{width}w.{suffix}"
+        return filename
+
+    def _calculate_height(self, width):
+        """Calculate height of resized image with given width"""
+        width_percent = (width / float(self.image.size[0]))
+        height = int((float(self.image.size[1]) * float(width_percent)))
```

Comment:
Do you not want to round this instead of always flooring it maybe?
That one pixel probably doesn't matter though I guess :D

Reply:
Haha this took me a while :D 
I think you read that wrong. I am converting to float not flooring.

Reply:
But thinking about it all the float casts are pretty much useless.
I refactored to:
```
width_percent = width / self.image.size[0]
height = int(self.image.size[1] * width_percent)
```

Reply:
I'll rename `width_percent` to `width_ratio`.